src/machine/asm/mod.rs

```diff
@@ -198,15 +290,17 @@ impl Memory for Box<AsmCoreMachine> {
     }
 
     fn execute_load16(&mut self, addr: u64) -> Result<u16, Error> {
-        let page_indices = get_page_indices(addr, 2)?;
-        check_permission(self, &page_indices, FLAG_EXECUTABLE)?;
-        self.load16(&(addr)).map(|v| v as u16)
+        check_memory_executable(self, addr, 2)?;
+        Ok(LittleEndian::read_u16(
+            &self.memory[addr as usize..addr as usize + 2],
+        ))
     }
 
     fn execute_load32(&mut self, addr: u64) -> Result<u32, Error> {
-        let page_indices = get_page_indices(addr, 4)?;
-        check_permission(self, &page_indices, FLAG_EXECUTABLE)?;
-        self.load32(&(addr)).map(|v| v as u32)
+        check_memory_executable(self, addr, 4)?;
+        Ok(LittleEndian::read_u32(
+            &self.memory[addr as usize..addr as usize + 4],
+        ))
     }
 
     fn load8(&mut self, addr: &u64) -> Result<u64, Error> {
```

Comment:
It would be great if you can also modify load8/16/32/64

Reply:
resolved, please review again.